routing/payment_lifecycle.go

```diff
@@ -724,6 +724,11 @@ func (p *paymentLifecycle) amendFirstHopData(rt *route.Route) error {
 	// value.
 	rt.FirstHopWireCustomRecords = p.firstHopCustomRecords
 
+	var firstHop route.Vertex
```

Comment:
So I think because the function is called `amendFirstHopData `, shouldn't we error out at the beginning of the function if:

```
if len(rt.Hops) == 0
```

If there is no route, why do we bother creating the FirstHopData in the first place.

Reply:
There can be keysend payments, which aren't invoice based and therefore don't have route hints attached. So this logic is okay the way it is.

Reply:
hmm but also keysends have a route which they need to follow I don't think we are talking about route hints here or ?

```
err = p.amendFirstHopData(rt)
```

the rt is a route not a route hint.

Reply:
Ah, yeah, my bad. This is the actual route that is going to be attempted, not the route hints. I guess your comment stands then :+1: